src/message.c

```diff
@@ -787,44 +787,103 @@ static void send_handler(void *context, const unsigned char *start, int length)
     pn_link_send(pnl, (const char*) start, length);
 }
 
+/**
+ * Returns true if the data backing the iterator matches any of the router specific annotations like "x-opt-qd.ingress" or
+ * "x-opt-qd.trace" or "x-opt-qd.to" or "x-opt-qd.phase
+ */
+static bool qd_message_is_router_annotation(qd_field_iterator_t *iter)
+{
+    bool is_router_annotation = true;
+    int i = 0;
+    while(! qd_field_iterator_end(iter)) {
```

Comment:
I believe this function is not needed.   But for the review, there is a bug in this algorithm.  It will return 'true' if the iterator field is blank or a subset of the prefix.  For example, a key of 'x-o' will be incorrectly identified as a router annotation.